base/libdl.jl

```diff
@@ -334,7 +334,7 @@ struct LazyLibraryPath
     LazyLibraryPath(pieces::Vector) = new(pieces)
 end
 LazyLibraryPath(args...) = LazyLibraryPath(collect(args))
-Base.string(llp::LazyLibraryPath) = joinpath(string.(llp.pieces)...)
+Base.string(llp::LazyLibraryPath) = joinpath(string.(llp.pieces)...)::String
```

Comment:
Not sure this is a good idea, but if you want to do it, it should be a return type declaration, so
```suggestion
Base.string(llp::LazyLibraryPath)::String = joinpath(string.(llp.pieces)...)
```

Reply:
> Not sure this is a good idea

Why would it be anything but good? There's basically no downside. In case the check is useless, it will get optimized out, otherwise it's useful because it helps contain type-instability.

> if you want to do it, it should be a return type declaration

Why? As far as I see the expression `joinpath(string.(llp.pieces)...)` is required to be a `String`, not just convertible to `String`, so I don't see a good reason for introducing conversion.

Reply:
I agree with @nsajko